app/Controllers/CtrlSolution.php

```diff
@@ -98,36 +134,67 @@ public function updateSolution(string $msId)
             ];
         }
 
-        return redirect()->to('/admin/soluciones')->with('response', $response);
+        return redirect()->to($this->MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
     }
 
     public function deleteSolution()
     {
         try {
-            $msId                     = $this->request->getPost()['msId'];
-            $measurementSolutionModel = new MeasurementSolutionModel();
+            $msModel = new MeasurementSolutionModel();
+
+            $msData = $msModel->getMeasurementSolutionDataWithFiles($this->request->getPost()['msId']);
 
-            $isDeleted = $measurementSolutionModel->delete($msId);
+            $isDeleted = (new MeasurementSolutionModel())->deleteMeasuremenSolution($msData);
 
             if (! $isDeleted) {
                 throw new Exception();
             }
 
+            $this->deleteMsFiles([$msData['msIcon'], $msData['msImage']]);
+
             $response = [
                 'title'   => 'Eliminación exitosa',
-                'message' => 'Se ha elimnado la solución de medición correctamente',
+                'message' => 'Se ha eliminado la solución de medición correctamente',
                 'type'    => 'success',
             ];
 
-            return redirect()->to('/admin/soluciones')->with('response', $response);
+            return redirect()->to($this->MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
         } catch (Throwable $th) {
             $response = [
                 'title'   => 'Oops! Ha ocurrido un error.',
                 'message' => 'Ha ocurrido un error al tratar de eliminar la solución de medición, por favor intente nuevamente.',
                 'type'    => 'error',
             ];
 
-            return redirect()->to('/admin/soluciones')->with('response', $response);
+            return redirect()->to($this->MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
```

Comment:
Update this constant 
```suggestion
            return redirect()->to(self::MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
```